grails-app/migrations/views/cycle-count-session.sql

```diff
@@ -35,48 +35,60 @@ SELECT
 
     # Stuff we need to include now or wait for the materialized view
     -- Negative item count
+    -- NULL as negative_item_count,
     (SELECT SUM(CASE WHEN pai.quantity_on_hand < 0 THEN 1 ELSE 0 END)
      FROM product_availability pai
      WHERE pai.product_id = product_availability.product_id
-       AND pai.location_id = product_availability.location_id)                       as negative_item_count,
+       AND pai.location_id = product_availability.location_id
+     group by pai.product_id, pai.location_id)                                       as negative_item_count,
 
     -- Date last counted
-    (select max(date_counted)
-     from product_count_history
-     where product_count_history.product_id = product_availability.product_id
-       and product_count_history.inventory_id = location.inventory_id
-     group by inventory_id, product_id)                                              as date_last_count,
-
-    NULL                                                                             as date_next_count,
+    -- NULL as date_last_count,
+    cycle_count_metadata.date_counted                                                as date_last_count,
```

Comment:
This small change (removing subquery and pulling date counted from this metadata view) carved off about 3-4 seconds from the query.